pkg/metricsstore/metricsstore.go

```diff
@@ -71,6 +74,20 @@ func init() {
 		Help:      "represents the number of proxies connected to OSM controller",
 	})
 
+	defaultMetricsStore.ProxyConfigUpdateTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsRootNamespace,
+			Subsystem: "proxy",
+			Name:      "config_update_time",
+			Buckets:   []float64{.1, .25, .5, 1, 2.5, 5, 10, 20, 40, 90},
+			Help:      "Histogram to track time spent for proxy configuration",
+		},
+		[]string{
+			"proxyName",    // proxyName is the common name of the proxy
+			"resourceType", // identifies a typeURI resource
+			"success",      // further labels if the operation succeeded or not
```

Comment:
snake_case for labels and metric names.

```suggestion
			"proxy_cn",    // proxy_cn is the common name of the proxy
			"resource_type", // identifies a typeURI resource
			"success",      // further labels if the operation succeeded or not
```